app/src/main/java/ru/prometeydev/movie/FragmentMoviesList.kt

```diff
@@ -1,27 +1,30 @@
 package ru.prometeydev.movie
 
+import android.content.Context
 import android.content.res.Configuration
 import android.os.Bundle
+import android.os.Parcelable
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import androidx.core.content.ContextCompat
 import androidx.fragment.app.Fragment
 import androidx.recyclerview.widget.GridLayoutManager
 import androidx.recyclerview.widget.RecyclerView
-import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.*
 import ru.prometeydev.movie.data.adapters.MoviesAdapter
-import ru.prometeydev.movie.data.domain.MoviesDataSource
-import ru.prometeydev.movie.data.models.Movie
+import ru.prometeydev.movie.data.Movie
+import ru.prometeydev.movie.data.loadMovies
 
 class FragmentMoviesList : Fragment() {
 
-    private var recycler: RecyclerView? = null
+    private var scope = CoroutineScope(Dispatchers.Default + Job())
 
-    private var movies: List<Movie>? = null
+    private var recycler: RecyclerView? = null
 
     private var spanCount = VERTICAL_SPAN_COUNT
 
+    private var movies: List<Movie> = listOf()
```

Comment:
Why store movies?

Reply:
Yes, you don't need to store it. Download and bind immediately.